From ffc66f9c8477b99d1fd70eb35bae5cc0057afff8 Mon Sep 17 00:00:00 2001
From: Sergio Pena <sergio.pena@cloudera.com>
Date: Sun, 18 Oct 2015 18:39:21 -0500
Subject: [PATCH 0347/1164] CDH-32299: HIVE-12188: DoAs does not work properly
 in non-kerberos secured HS2 (Chaoyu Tang,
 reviewed by Sergio Pena)

Change-Id: I3114a2ed627e16dabb71ddfca48f6c9dfd4254ae
---
 .../hive/service/cli/thrift/ThriftCLIService.java  |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/service/src/java/org/apache/hive/service/cli/thrift/ThriftCLIService.java b/service/src/java/org/apache/hive/service/cli/thrift/ThriftCLIService.java
index 67bc778..926f18f 100644
--- a/service/src/java/org/apache/hive/service/cli/thrift/ThriftCLIService.java
+++ b/service/src/java/org/apache/hive/service/cli/thrift/ThriftCLIService.java
@@ -434,14 +434,11 @@ SessionHandle getSessionHandle(TOpenSessionReq req, TOpenSessionResp res)
 
   private String getDelegationToken(String userName)
       throws HiveSQLException, LoginException, IOException {
-    if (userName == null || !cliService.getHiveConf().getVar(ConfVars.HIVE_SERVER2_AUTHENTICATION)
-        .equalsIgnoreCase(HiveAuthFactory.AuthTypes.KERBEROS.toString())) {
-      return null;
-    }
     try {
       return cliService.getDelegationTokenFromMetaStore(userName);
     } catch (UnsupportedOperationException e) {
       // The delegation token is not applicable in the given deployment mode
+      // such as HMS is not kerberos secured 
     }
     return null;
   }
-- 
1.7.9.5

